src/streamlink/stream/hls_playlist.py

```diff
@@ -238,9 +238,19 @@ def parse(self, data):
         self.state = {}
         self.m3u8 = M3U8()
 
-        for lineno, line in enumerate(filter(bool, data.splitlines())):
-            self.parse_line(lineno, line)
+        lines = iter(filter(bool, data.splitlines()))
+        try:
+            line = next(lines)
+        except StopIteration:
+            return self.m3u8
+        else:
+            if not line.startswith("#EXTM3U"):
+                raise ValueError("Missing #EXTM3U header")
 
+        parse_line = self.parse_line
+        for line in lines:
+            parse_line(line)
+            
```

Comment:
@Billy2011 
**can you remove these spaces here,**
did not see it before because my editor removes them automatically.

This is the only issue left and should be good to merge after this.

I did a small `speed` test on an old FireTV,
it has still some small speed improvements in thousandths of a second not much but still.